src/Compilers/Core/Portable/InternalUtilities/StackGuard.cs

```diff
@@ -18,11 +17,9 @@ public static void EnsureSufficientExecutionStack(int recursionDepth)
             }
         }
 
-        // TODO (DevDiv workitem 966425): Replace exception name test with a type test once the type 
-        // is available in the PCL
         public static bool IsInsufficientExecutionStackException(Exception ex)
```

Comment:
>public static bool IsInsufficientExecutionStackException(Exception ex) [](start = 8, length = 70)

I think we should remove this method and replace code like
```
catch (Exception ex) when (StackGuard.IsInsufficientExecutionStackException(ex))
```

with
```
catch (InsufficientExecutionStackException ex)
```

Reply:
Also might want to add a comment to the method above, that it throws ```InsufficientExecutionStackException``` when available stack size is not sufficient.

---
In reply to: [234307653](https://github.com/dotnet/roslyn/pull/30817#discussion_r234307653) [](ancestors = 234307653)

Reply:
That a good point. I have rebased and updated with your suggestions.